### PR TITLE
Fix open pull request on GitHub button

### PR DESCRIPTION
**Pull request by abrl91**
Issue Number: #3811 

## PR Details
- when the user is in the context of a service (resource), the open with GitHub button opens the last pull request (last build)
- in a context of  project, the open with GitHub button opens the repository URL that is connected to the **project**.

## PR Checklist
- [ ] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

**Comment by abrl91** [comment]
"in a context of project, the open with GitHub button opens the repository URL that is connected to the project"

@GreenMachine01 @yuval-hazaz 
This is the decision I have made, but we may think about it.
A project can have more than one resource. Each resource can be connected to a different git repository. 
When the user clicks on the build button, it creates a commit with x number of builds, a build to every resource on the project. 
We can decide that the button opens the last build no matter what, but then it can cause strange behavior. For example, the user is on one of his services pages, and clicks on the open with GitHub button, but the current service is not the one that was built last, and as a result, the wrong pull request will be opened

File: packages/amplication-client/src/Workspaces/WorkspaceFooter.tsx [hunk 1]
```diff
@@ -86,13 +102,23 @@ const WorkspaceFooter: React.FC<{}> = () => {
...
+              </a>
+            ) : (
+              <a
+                className={`${CLASS_NAME}__gh-link`}
+                href={gitRepositoryUrl}
+                target="github"
+              >
+                Open With GitHub
+              </a>
+            )}
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Workspaces/WorkspaceFooter.tsx:121]: use only one <a> and make sure that value returned from githubUrl is equal to githubUrl when needed

File: packages/amplication-client/src/Workspaces/WorkspaceFooter.tsx [hunk 2]
```diff
@@ -75,6 +76,21 @@ const WorkspaceFooter: React.FC<{}> = () => {
     />
   );
 
+  const githubUrl = useMemo(() => {
+    if (!lastResourceBuild?.action?.steps?.length) {
+      return null;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Workspaces/WorkspaceFooter.tsx:81]:
```suggestion
      return gitRepositoryUrl;
```

File: packages/amplication-client/src/Workspaces/WorkspaceFooter.tsx [hunk 3]
```diff
@@ -75,6 +76,21 @@ const WorkspaceFooter: React.FC<{}> = () => {
...
+    }
+    const stepGithub = lastResourceBuild?.action.steps.find(
+      (step) => step.name === PUSH_TO_GITHUB_STEP_NAME
+    );
+
+    const log = stepGithub?.logs?.find(
+      (log) => !isEmpty(log.meta) && !isEmpty(log.meta.githubUrl)
+    );
+
+    return log?.meta?.githubUrl || null;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Workspaces/WorkspaceFooter.tsx:91]:
```suggestion
    return log?.meta?.githubUrl || gitRepositoryUrl;
```

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**